Jint.Tests/Runtime/ModuleTests.cs

```diff
@@ -0,0 +1,195 @@
+        }
+
+        [Fact]
+        public void ShouldPropagateThrowStatementThroughJavaScriptImport()
+        {
+            _engine.DefineModule(@"throw new Error('imported successfully');", "imported-module");
+            _engine.DefineModule(@"import 'imported-module';", "my-module");
+
+            var exc = Assert.Throws<JavaScriptException>(() => _engine.ImportModule("my-module"));
+            Assert.Equal("imported successfully", exc.Message);
+            Assert.Equal("imported-module", exc.Location.Source);
+        }
+
+        [Fact]
+        public void ShouldDefineModuleFromJsValue()
+        {
+            _engine.DefineModule("value", JsString.Create("hello world"), "my-module");
+            var ns = _engine.ImportModule("my-module");
+
+            Assert.Equal("hello world", ns.Get("value").AsString());
+        }
+
+        [Fact]
+        public void ShouldDefineModuleFromClrInstance()
+        {
+            _engine.DefineModule("value", new ImportedClass { Value = "instance value" }, "imported-module");
+            _engine.DefineModule(@"import { value } from 'imported-module'; export const exported = value.value;", "my-module");
+            var ns = _engine.ImportModule("my-module");
+
+            Assert.Equal("instance value", ns.Get("exported").AsString());
+        }
+
+        [Fact]
+        public void ShouldDefineModuleFromClrType()
+        {
+            _engine.DefineModule<ImportedClass>("imported-module");
+            _engine.DefineModule(@"import { ImportedClass } from 'imported-module'; export const exported = new ImportedClass().value;", "my-module");
+            var ns = _engine.ImportModule("my-module");
+
+            Assert.Equal("hello world", ns.Get("exported").AsString());
+        }
+
+        [Fact]
+        public void ShouldAllowExportMultipleImports()
+        {
+            _engine.DefineModule("value1", JsNumber.Create(1), "@mine/import1");
+            _engine.DefineModule("value2", JsNumber.Create(2), "@mine/import2");
+            _engine.DefineModule("export * from '@mine/import1'; export * from '@mine/import2'", "@mine");
+            _engine.DefineModule(@"import { value1, value2 } from '@mine'; export const result = `${value1} ${value2}`", "app");
+            var ns = _engine.ImportModule("app");
+
+            Assert.Equal("1 2", ns.Get("result").AsString());
+        }
+
+        private class ImportedClass
+        {
+            public string Value { get; set; } = "hello world";
+        }
+
+#if(NET6_0_OR_GREATER)
+
+        [Fact]
+        public void CanLoadModuleImportsFromFiles()
+        {
+            var engine = new Engine(options => options.EnableModules(GetBasePath()));
+            engine.DefineModule("import { User } from './modules/user.js'; export const user = new User('John', 'Doe');", "my-module");
+            var ns = engine.ImportModule("my-module");
+
+            Assert.Equal("John Doe", ns["user"].AsObject()["name"].AsString());
```

Comment:
Isn't it sufficient to let users access the module properties from the engine since the module was loaded? `engine.GetValue("user")` just to reduce the amounts of ways to do something, which could lead to inconsistencies, or discrepancies in behavior. So `ImportModule` would return nothing. (or Engine)